github/github.go

```diff
@@ -1102,6 +1113,12 @@ type rateLimitCategory uint8
 const (
 	coreCategory rateLimitCategory = iota
 	searchCategory
+	graphqlCategory                   //nolint:deadcode,varcheck
+	integrationManifestCategory       //nolint:deadcode,varcheck
+	sourceImportCategory              //nolint:deadcode,varcheck
+	codeScanningUploadCategory        //nolint:deadcode,varcheck
+	actionsRunnerRegistrationCategory //nolint:deadcode,varcheck
+	scimCategory                      //nolint:deadcode,varcheck
```

Comment:
Since `TestClient_rateLimits` does not pass, I have defined unused constants.

Reply:
Instead of doing this, we actually need to use these values in the `category` function (lines 1127-1134 below) and in the `RateLimits` method (lines 1154-1163 below).

Reply:
> in the RateLimits method (lines 1154-1163 below).

fixed https://github.com/google/go-github/pull/2340/commits/a6c7bcd7f8602717e83b98d98444c9da8a7920d8

> in the category function (lines 1127-1134 below)

I could not figure out how to fix the `category` function for the following reasons...

> Since there is no documentation for source_import, code_scanning_upload, actions_runner_registration, and scim, I did not know which endpoints had the rate limit set, so I did not fix category function.
https://github.com/google/go-github/pull/2340#issue-1213489146